src/SurveyService.Validation/UserAnswer/CreateUserAnswerRequestValidator.cs

```diff
@@ -0,0 +1,43 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.UserAnswer;
+using LT.DigitalOffice.SurveyService.Validation.UserAnswer.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.SurveyService.Validation.UserAnswer;
+
+public class CreateUserAnswerRequestValidator : AbstractValidator<CreateUserAnswerRequest>, ICreateUserAnswerRequestValidator
+{
+  public CreateUserAnswerRequestValidator(
+    IOptionRepository _optionRepository,
+    IHttpContextAccessor _httpContextAccessor)
+  {
+    RuleFor(request => new
+    {
+      optionsIds = request.OptionIds,
+      options = _optionRepository.GetByIdsAsync(request.OptionIds).Result
+    })
+      .Cascade(CascadeMode.Stop)
+      .Must(x => x.optionsIds.Any())
+      .WithMessage("There should be at least one option in request.")
+      .Must(x => x.optionsIds.Count == x.options.Count)
+      .WithMessage("Some option ids don't exist.")
+      .Must(x => x.options.Select(o => o.Question).Distinct().Count() == 1
+        || x.options.Select(o => o.Question).Distinct().Select(q => q.GroupId).Distinct().Count() == 1)
```

Comment:
it catchs options from different groups, but it doesn't catch options from different single questions

Reply:
Done